src/main/java/org/jenkinsci/plugins/googleplayandroidpublisher/Util.java

```diff
@@ -46,32 +49,22 @@
 
     /** @return The version of this Jenkins plugin, e.g. "1.0" or "1.1-SNAPSHOT" (for dev releases). */
     public static String getPluginVersion() {
-        final String version = Jenkins.getInstance().getPluginManager().whichPlugin(Util.class).getVersion();
-        int index = version.indexOf(' ');
-        return (index == -1) ? version : version.substring(0, index);
+        return sJenkins.getPluginVersion();
     }
 
     /** @return The application ID of the given APK file. */
     public static String getApplicationId(FilePath apk) throws IOException, InterruptedException {
-        return apk.act(new MasterToSlaveFileCallable<String>() {
-            public String invoke(File f, VirtualChannel channel) throws IOException {
-                return getApkMetadata(f).getPackageName();
-            }
-        });
+        return sJenkins.actOnPath(apk, file -> sAndroid.getApkPackageName(file));
```

Comment:
I'll test it on a newer Jenkins and update as needed

Reply:
Don't worry about it.. the tests pass on a newer Jenkins because it's not doing any serialisation to a "remote" build agent — the build steps are all being executed on the master, so there's no serialisation failures.  I sent you some more details via email :)

Reply:
I'm getting a similar serialization error when I try adding a test for a remote slave, but it's not even related to the callable. Looks like ApkUploadTask also can't be serialized:

```
Caused: java.io.IOException: Unable to serialize org.jenkinsci.plugins.googleplayandroidpublisher.ApkUploadTask@1be57b8d
	at hudson.remoting.UserRequest.serialize(UserRequest.java:201)
	at hudson.remoting.UserRequest.<init>(UserRequest.java:64)
	at hudson.remoting.Channel.call(Channel.java:828)
	at hudson.FilePath.act(FilePath.java:1082)
	at org.jenkinsci.plugins.googleplayandroidpublisher.ApkPublisher.publishApk(ApkPublisher.java:374)
	at org.jenkinsci.plugins.googleplayandroidpublisher.ApkPublisher.perform(ApkPublisher.java:194)
```

Reply:
Oh I see now why:

    java.io.NotSerializableException: org.mockito.internal.creation.DelegatingMethod